### ISSUE TITLE
My Home: Some buttons in Quick links section aren't accessible

### ISSUE THREAD

**Issue by p-jackson** [Automattic/wp-calypso#54483]
### Steps to reproduce the behavior

1. Go to My Home and expand quick links
2. Navigate the quicklinks section using tab navigation, or try using those buttons with a screen reader (e.g. VoiceOver in macOS)

#### What I expected to happen

Quick links looks like a collection of links and buttons, so I'd expect them all to be "clickable" with something other than a mouse, and for a screen reader to announce them.

#### What actually happened

The following buttons aren't accessible:
- Edit homepage
- Add a page
- Write blog post
- Change theme
- Add a domain

<img width="328" alt="ql" src="https://user-images.githubusercontent.com/1500769/125373893-03240500-e3da-11eb-8063-82b9e8c246ea.png">

They aren't tab navigable https://user-images.githubusercontent.com/1500769/121634557-e04fb980-cad8-11eb-959d-d76f8ccc6a71.mov

If you take a look at these elements in Firefox's accessibility tools you can see they have no label and there's a keyboard error.

<img width="1282" alt="a11ytools" src="https://user-images.githubusercontent.com/1500769/125373904-06b78c00-e3da-11eb-8ade-d597b84cd9de.png">

---

It looks like these buttons aren't using the `<button>` tag at all, and are instead `<div>`s with `onClick` handlers. Switching to using a semantic `<button>` tag should fix these issues.

#### Context

One of a number of a11y issues in #53617

**Comment by p-jackson** [comment]
Just setting up some work for when you join the team @arthur791004. Don't start this yet, just wanted to assign it to you so that no one else picked it up first 😅

**Comment by autumnfjeld** [comment]
@arthur791004 should this issue be closed? :)

**Comment by taipeicoder** [comment]
This seems to be taken care of. Closing :-)
